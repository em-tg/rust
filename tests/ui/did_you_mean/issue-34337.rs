//@ check-pass
fn get(key: &mut String) { }

fn main() {
    let mut v: Vec<String> = Vec::new();
    let ref mut key = v[0];
    get(&mut key); //~ WARNING cannot borrow
    //~| HELP try removing `&mut` here
}
