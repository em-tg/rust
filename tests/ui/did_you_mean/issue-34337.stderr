warning[E0596]: cannot borrow `key` as mutable, as it is not declared as mutable
  --> $DIR/issue-34337.rs:7:9
   |
LL |     get(&mut key);
   |         ^^^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: try removing `&mut` here
   |
LL -     get(&mut key);
LL +     get(key);
   |

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
